### PR TITLE
Documentation: redis update

### PR DESCRIPTION
**Pull request by boesing**
|    Q          |   A
|-------------- | ------
| Documentation | yes

### Description

Optimized some redis documentation and added initial `RedisCluster` adapter documentation.

`RedisCluster` adapter is provided in laminas/laminas-cache-storage-adapter-redis#13

### CODE REVIEW

File: docs/book/storage/adapter.md [hunk 1]
```diff
@@ -782,6 +784,47 @@ Name | Data Type | Default Value | Description
 - Associative array: `['host' => <host>[, 'port' => <port>[, 'timeout' => <timeout>]]]`
 - List: `[<host>[, <port>, [, <timeout>]]]`
 
+## RedisCluster Adapter
+
+`Laminas\Cache\Storage\Adapter\RedisCluster` stores cache items over the redis cluster protocol
+using the PHP extension [redis](https://github.com/phpredis/phpredis).
```
Comment (froschdesign) [inline docs/book/storage/adapter.md]: I suggest to add a link for the cluster support. (the link goes to the readme, not to any blob file in the repository)
```suggestion
`Laminas\Cache\Storage\Adapter\RedisCluster` stores cache items over the [Redis cluster](https://github.com/phpredis/phpredis#redis-cluster-support) protocol
using the PHP extension [PhpRedis](https://github.com/phpredis/phpredis).
```

File: docs/book/storage/adapter.md [hunk 2]
```diff
@@ -741,10 +741,12 @@ Name | Data Type | Default Value | Description
 ## Redis Adapter
 
 `Laminas\Cache\Storage\Adapter\Redis` stores cache items over the redis protocol
```
Comment (froschdesign) [inline docs/book/storage/adapter.md]: "[Redis](https://redis.io)" is a proper / product name therefore I suggest the official spelling.
```suggestion
`Laminas\Cache\Storage\Adapter\Redis` stores cache items over the [Redis](https://redis.io) protocol
```

File: docs/book/storage/adapter.md [hunk 3]
```diff
@@ -741,10 +741,12 @@ Name | Data Type | Default Value | Description
...
-using the PHP extension [redis](https://github.com/nicolasff/phpredis).
+using the PHP extension [redis](https://github.com/phpredis/phpredis).
```
Comment (froschdesign) [inline docs/book/storage/adapter.md]: The extension is installed with "redis", but the official name seems to be "[PhpRedis](https://github.com/phpredis/phpredis#phpredis)".
```suggestion
using the PHP extension [PhpRedis](https://github.com/phpredis/phpredis).
```

**Review by froschdesign — CHANGES_REQUESTED** [review]

**Review by froschdesign — APPROVED** [review]

**Comment by boesing** [comment]
Thanks, @froschdesign!

**Pull request merged**